src/iterative_sorting/iterative_sorting.py

```diff
@@ -20,7 +19,11 @@ def selection_sort( arr ):
 
 # TO-DO:  implement the Bubble Sort function below
 def bubble_sort( arr ):
-
+    length = len(arr)
+    for i in range(length):
+        for a in range(0, length -i -1):
+            if arr[a] > arr[a+1]:
+                arr[a],arr[a+1] = arr[a+1],arr[a]
```

Comment:
this code is really nice and does the work really well. Traditionally bubble sort uses a True/False variable to keep tabs on whether everything is sorted. This is important because while your code solves the problems, if you were given a completely sorted list or even a partly sorted list, your code would go through it all completely as if it were an unsorted list before determining it was sorted. So your best case and worst case time complexity are the same. If you used the True/False variable to determine whether something was sorted it would cut down on some potential runs through the code.

something like this:

made_changes = True
while made_changes:
made_changes = False
(for loop)
(comparing)
(swap if needed and then changing made_changes to True so the loop continues)